File: go/1901.go
package main

func findPeakGrid(mat [][]int) []int {
	l, r := 0, len(mat)-1
	for l < r {
		mid := l + (r-l)/2
		idx := indexOfMax(mat[mid])
		if mat[mid+1][idx] > mat[mid][idx] {
			l = mid + 1
		} else {
			r = mid
		}
	}
	return []int{l, indexOfMax(mat[l])}
}

func indexOfMax(a []int) (idx int) {
	for i, x := range a {
		if x > a[idx] {
			idx = i
		}
	}
	return
}

File: python/1901.py
from typing import List

class Solution:
    def findPeakGrid(self, mat: List[List[int]]) -> List[int]:
        l, r = 0, len(mat) - 1
        while l < r:
            i = (l + r) // 2
            mx = max(mat[i])
            if mat[i+1][mat[i].index(mx)] > mx:
                l = i + 1
            else:
                r = i
        return [l, mat[l].index(max(mat[l]))]